avocado/utils/software_manager.py

```diff
@@ -158,6 +144,11 @@ def __getattr__(self, name):
         self._init_on_demand()
         return self.backend.__getattribute__(name)
 
+    @classmethod
+    def _get_backends(cls):
```

Comment:
why is this private? You're aalready acessing it from the outside...

Reply:
Hello @apahim! My rationale is: `_get_backends` should be internal to the module, so I don't want users accessing it as an API. Unless others think it should be API exposed?

Reply:
Then it should live in the module, as a function, not in the class...

Reply:
> Then it should live in the module, as a function, not in the class...

Which means that the end result would be more like:

```python
PACKAGE_MANAGERS = {'apt-get': AptBackend, 'yum': YumBackend, 'dnf': DnfBackend, 'zypper': ZypperBackend}

...

for high_level_pm in PACKAGE_MANAGERS:
```

Reply:
@apahim, 
Because I spent a few years working with Java, it makes complete sense to leave this a class method. But I understand your reasoning too, so I will going to implement your suggestion on v2.